### PR TITLE
Add in identity bits

### PR DESCRIPTION
**Pull request by s3364859**
@6nop @s3364942 @s3494110 @s3482247 

In this PR:
- Identity data bits
- Identity object model
- Hashing function

identity

### CODE REVIEW

File: CPT331.Core/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 	public static class StringExtensions
 	{
+		private static SHA1CryptoServiceProvider _sha1CryptoServiceProvider;
+
 		public static bool EqualsIgnoreCase(this string value, string compareTo)
 		{
 			return value.Equals(compareTo, StringComparison.OrdinalIgnoreCase);
 		}
+
+		public static string Hash(this string value)
```
Comment (s3364859) [inline CPT331.Core/Extensions/StringExtensions.cs]: Hash this!

File: CPT331.Core/ObjectModel/User.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
+﻿#region Using References
+
+using System;
+
+#endregion
+
+namespace CPT331.Core.ObjectModel
+{
+	public class User
```
Comment (s3364859) [inline CPT331.Core/ObjectModel/User.cs]: User object model type.

File: CPT331.Data/UserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				user = SqlMapper
+					.Query(sqlConnection, "Registration.spGetUserByID", new { ID = id }, commandType: CommandType.StoredProcedure)
+					.Select(m => new User(m.DateCreatedUtc, m.DateUpdatedUtc, m.ID, m.IsActive, m.IsDeleted, m.Password, m.Username))
+					.FirstOrDefault();
+			}
+
+			return user;
+		}
+
+		public static User GetUserByUsername(string username)
```
Comment (s3364859) [inline CPT331.Data/UserRepository.cs]: Probably use this for login authentication.

File: CPT331.Web/Attributes/AdminAuthorizeAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Routing;
+
+namespace CPT331.Web.Attributes
+{
+    public class AdminAuthorizeAttribute : System.Web.Mvc.AuthorizeAttribute
```
Comment (6nop) [inline CPT331.Web/Attributes/AdminAuthorizeAttribute.cs]: Code previously thrown into Skype, so I saw no harm it throwing it in.

File: CPT331.Web/Controllers/AccountController.cs [hunk 5]
```diff
@@ -4,24 +4,24 @@
 using System.Web;
 using System.Web.Mvc;
 using CPT331.Web.Models.Account;
+using CPT331.Web.Attributes;
 
 namespace CPT331.Web.Controllers
 {
+    [AdminAuthorize]
```
Comment (6nop) [inline CPT331.Web/Controllers/AccountController.cs]: Locked down class, opened up Login actions only

File: CPT331.Web/Controllers/AccountController.cs [hunk 6]
```diff
@@ -4,24 +4,24 @@
...
             ViewBag.Title = "Admin";
             return View();
         }
 
-        [AcceptVerbs(HttpVerbs.Post)]
+        [AllowAnonymous]
+        [HttpPost]
         public ActionResult Login(string loginName, string password)
         {
             if ("administrator".Equals(loginName) && "Rgx53r$t5r".Equals(password))
```
Comment (6nop) [inline CPT331.Web/Controllers/AccountController.cs]: IMG Nothing to see here

Implementation left for Angus to sort out later on...

Comment (s3364859) [inline CPT331.Web/Controllers/AccountController.cs]: Yep - have a look at the new hashing code.

Also "mystring".Equals("somethingelse") looks weird to me - anyone else? The idea there is to call the == operator and let the implementation choose which Equals method to call.

Comment (6nop) [inline CPT331.Web/Controllers/AccountController.cs]: Yes, I agree. I also think in this context, I prefer to have the literal value on the right.

* ```loginName.Equals("administrator")``` is a nicer convention than ```"administrator".Equals(loginName)```

* And ```loginName == "administrator"``` is easier to understand ```loginName.Equals("administrator")```.

The use of Equals() is too Java for my liking :)

Comment (s3364859) [inline CPT331.Web/Controllers/AccountController.cs]: What's a 'Java'? Never heard of it.

Comment (s3494110) [inline CPT331.Web/Controllers/AccountController.cs]: Just on this again, yes this was a bad habit from Java where everything is a reference comparison. I need to put that in my past. :D @s3364859 @6nop

**Pull request merged**